internal/alloc/alloc_windows.go

```diff
@@ -0,0 +1,92 @@
+//go:build windows
+
+package alloc
+
+import (
+	"fmt"
+	"math"
+	"syscall"
+	"unsafe"
+
+	"github.com/tetratelabs/wazero/experimental"
+)
+
+var (
+	kernel32         = syscall.NewLazyDLL("kernel32.dll")
+	procVirtualAlloc = kernel32.NewProc("VirtualAlloc")
+	procVirtualFree  = kernel32.NewProc("VirtualFree")
+)
+
+const (
+	windows_MEM_COMMIT     uintptr = 0x00001000
+	windows_MEM_RESERVE    uintptr = 0x00002000
+	windows_MEM_RELEASE    uintptr = 0x00008000
+	windows_PAGE_READWRITE uintptr = 0x00000004
+
+	// https://cs.opensource.google/go/x/sys/+/refs/tags/v0.20.0:windows/syscall_windows.go;l=131
+	windows_PAGE_SIZE uint64 = 4096
+)
+
+func Allocator() experimental.MemoryAllocator {
+	return experimental.MemoryAllocatorFunc(virtualAllocator)
+}
+
+func virtualAllocator(cap, max uint64) experimental.LinearMemory {
+	// Round up to the page size.
+	rnd := windows_PAGE_SIZE - 1
+	max = (max + rnd) &^ rnd
+	cap = (cap + rnd) &^ rnd
+
+	if max > math.MaxInt {
+		// This ensures int(max) overflows to a negative value.
```

Comment:
This comment could be tweaked: `unix.Mmap` uses signed `int` (hence my wording); Windows is using `uintptr` and `SIZE_T` (both unsigned).

The code still makes sense: it ensures that on 32-bit platforms a very large value overflows to either -1 (Unix) or `MaxUint32` (both of which cause a noisy error, rather than the silent corruption).